typing/uniqueness_analysis.ml

```diff
@@ -235,21 +235,41 @@ module Usage : sig
   (** Parallel composition *)
   val par : t -> t -> t
 end = struct
-  (* We have Unused (top) > Borrowed > Aliased > Unique > Error (bot).
+  (* [Usage.t] describes how a program uses a value:
 
      - Unused means unused
-     - Borrowed means read-only access confined to a region
-     - Aliased means read-only access that may escape a region. For example,
-     storing the value in a cell that can be accessed later.
-     - Unique means accessing the value as if it's the only pointer. Example
-     includes overwriting.
+     - Borrowed means read-only usage confined to a region
+     - Aliased means read-only usage that may escape a region. For example,
+     storing the value in a cell that can be used later.
+     - Unique means using the value as if it's the usage. Example includes
+       overwriting.
      - Error means error happens when composing usage.
 
-     Some observations:
-     - It is sound to relax mode towards Error. It grants the access more
-     "capability" and usually helps performance.
-       For example, relaxing borrowed to aliased allows code motion of
-       projections. Relaxing aliased to unique allows in-place update.
+     And we have Unused (top) > Borrowed > Aliased > Unique > Error (bot).
+
+     The analysis looks at a program, and infers the usage level of each
+     recursive sub-program wrt a value. In particular, it infers the usage
+     level of each Pexp_ident. The inferred usage level is compared against the
+     needed usage level (the [Typedtree.unique_use] in [Pexp_ident]), and errors
+     are raised if that fails.
+
+      For example:
+      [
+      let x = .. in
+      use_as_unique x;
+      use_as_unique x
+      ]
+
+      the [Typedtree.unique_use] of both [Pexp_ident x] would say "need unique",
+      but the analysis infers that [x] is used multiple times and thus each use
+      site can only use it as shared. Error is then raised.
+
+      It is sound for the analysis to infer a [Pexp_ident] to have a
```

Comment:
s/[Pexp_ident]/use

(since we also treat field reads this way)

Reply:
Good point. How about 
> .. infer a usage (such as [Pexp_ident]) to have a

Because I think having examples are more readable.

Reply:
I think I would prefer it if we explained at the top of the comment what a usage is and referred the reader to `[check_uniqueness_exp_as_value]`. Then we can simply write "usage" in the rest of the comment without extra examples.